pkg/notes/client/recordreplay/record_client.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recordreplay
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"sync"
+
+	"github.com/google/go-github/v28/github"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/release/pkg/notes/client"
+)
+
+type GitHubAPI string
+
+const (
+	GitHubAPIGetCommit                  GitHubAPI = "GetCommit"
+	GitHubAPIListCommits                GitHubAPI = "ListCommits"
+	GitHubAPIListPullRequestsWithCommit GitHubAPI = "ListPullRequestsWithCommit"
+	GitHubAPIGetPullRequest             GitHubAPI = "GetPullRequest"
+)
+
+func NewGithubNotesRecordClient(c *client.GithubNotesClient, recordDir string) client.Client {
+	return &githubNotesRecordClient{
+		client:      c,
+		recordDir:   recordDir,
+		recordState: map[GitHubAPI]int{},
+	}
+}
+
+type githubNotesRecordClient struct {
+	client      *client.GithubNotesClient
```

Comment:
I think that should be the interface, not the discrete type.

```suggestion
	client      *client.Client
```

Reply:
Fixed :+1: